source/code/plugin/health/health_kubernetes_resources.rb

```diff
@@ -0,0 +1,102 @@
+require 'singleton'
+
+module HealthModel
+    class HealthKubernetesResources
+
+        include Singleton
+        attr_accessor :node_inventory, :pod_inventory, :deployment_inventory
+        attr_reader :nodes, :pods, :workloads
+
+        def initialize
+            @node_inventory = []
+            @pod_inventory =  []
+            @deployment_inventory =  []
+            @nodes = []
+            @pods = []
+            @workloads = []
+            @log = HealthMonitorHelpers.get_log_handle
+        end
+
+        def get_node_inventory
+            return @node_inventory
+        end
+
+        def get_nodes
+            @nodes = []
+            @node_inventory['items'].each {|node|
+                if !@nodes.include?(node['metadata']['name'])
+                    @nodes.push(node['metadata']['name'])
+                end
+
+            }
+            return @nodes
+        end
+
+        def get_pod_inventory
+            return @pod_inventory
+        end
+
+        def get_pods
+            return @pods
+        end
+
+        def get_workload_names
+            @pods = []
+            workload_names = {}
+            deployment_lookup = {}
+            @deployment_inventory['items'].each do |deployment|
+                match_labels = deployment['spec']['selector']['matchLabels'].to_h
+                namespace = deployment['metadata']['namespace']
+                match_labels.each{|k,v|
+                    deployment_lookup["#{namespace}-#{k}=#{v}"] = "#{deployment['metadata']['namespace']}~~#{deployment['metadata']['name']}"
```

Comment:
is ~ a disallowed character for deployments?

Reply:
Yes. I checked